packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts

```diff
@@ -497,6 +529,212 @@ describe('Validation at REST level', () => {
+            'product-code': {
+              async: true,
+              validate: validateProductCode,
+            },
+          },
+        }),
+      );
+      after(() => app.stop());
+
+      it('validates name & code against custom ajv format', async () => {
+        const DATA = {
+          name: 'iPhone #9',
+          code: '123-ABC',
+        };
+        const res = await client
+          .post('/products')
+          .send(DATA)
+          .expect(422);
+
+        expect(res.body).to.eql({
+          error: {
+            statusCode: 422,
+            name: 'UnprocessableEntityError',
+            message:
+              'The request body is invalid. See error object `details` property for more info.',
+            code: 'VALIDATION_FAILED',
+            details: [
+              {
+                path: '/name',
+                code: 'errorMessage',
+                message: 'Name must not have special characters',
+                info: {
+                  errors: [
+                    {
+                      keyword: 'format',
+                      dataPath: '/name',
+                      schemaPath:
+                        '#/components/schemas/ProductWithCode/properties/name/format',
+                      params: {format: 'only-alphanumeric'},
+                      message: 'should match format "only-alphanumeric"',
+                    },
+                  ],
+                },
+              },
+              {
+                path: '/code',
+                code: 'errorMessage',
+                message: 'Invalid product code',
+                info: {
+                  errors: [
+                    {
+                      keyword: 'format',
+                      dataPath: '/code',
+                      schemaPath:
+                        '#/components/schemas/ProductWithCode/properties/code/format',
+                      params: {format: 'product-code'},
+                      message: 'should match format "product-code"',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        });
+      });
+    },
+  );
+
+  context(
+    'with request body validation options - {ajvCustomKeywords: {}}',
+    () => {
+      class ProductController {
+        @post('/products')
+        async create(
+          @requestBody({required: true}) data: ProductWithCode,
+        ): Promise<ProductWithCode> {
+          return new ProductWithCode(data);
+        }
+      }
+
+      before(() =>
+        givenAnAppAndAClient(ProductController, {
+          nullable: false,
+          compiledSchemaCache: new WeakMap(),
+          $data: true,
+          ajvKeywords: true,
+          ajvErrors: {
+            singleError: true,
+          },
+          ajvCustomFormats: {
+            'only-alphanumeric': /^[a-zA-Z0-9 ]+$/.source,
+            'product-code': {
+              async: true,
+              validate: async (code: string) => {
```

Comment:
Is there a way to restrict asynchronous validation for now ? Any benefits or necessity of having to support both ajv custom formats and asynchronous validation at the same time ?